cloud_export_to_parquet/data_trans_activities.py

```diff
@@ -0,0 +1,129 @@
+import json
+import uuid
+from dataclasses import dataclass
+from typing import List
+
+import aioboto3
+import pandas as pd
+import temporalio.api.export.v1 as export
+from google.protobuf.json_format import MessageToJson
+from temporalio import activity
+
+
+@dataclass
+class GetObjectKeysActivityInput:
+    bucket: str
+    path: str
+
+
+@dataclass
+class DataTransAndLandActivityInput:
+    export_s3_bucket: str
+    object_key: str
+    output_s3_bucket: str
+    write_path: str
+
+
+@activity.defn
+async def get_object_keys(activity_input: GetObjectKeysActivityInput) -> List[str]:
+    """Function that list objects by key."""
+    session = aioboto3.Session()
+    object_keys = []
+    async with session.client("s3") as s3:
+        response = await s3.list_objects_v2(
+            Bucket=activity_input.bucket, Prefix=activity_input.path
+        )
+
+        for obj in response.get("Contents", []):
+            object_keys.append(obj["Key"])
+        if len(object_keys) == 0:
+            raise FileNotFoundError(
+                f"No files found in {activity_input.bucket}/{activity_input.path}"
+            )
+
+    return object_keys
+
+
+@activity.defn
+async def data_trans_and_land(activity_input: DataTransAndLandActivityInput) -> str:
+    """Function that convert proto to parquet and save to S3."""
+    key = activity_input.object_key
+    data = await get_data_from_object_key(activity_input.export_s3_bucket, key)
+    activity.logger.info("Convert proto to parquet for file: %s", key)
+    parquet_data = convert_proto_to_parquet_flatten(data)
```

Comment:
I am concerned this is CPU bound work, is it expensive (i.e. can it take a lot of time blocking the entire asyncio event loop)?

I think `def` activities would be better than `async def` activities for code that does data translation and such.

Reply:
So while we do convert_proto_to_parquet_flatten, reading from s3 and writing to s3 are also I/O  happy operations, it would be benefit if we still keep it as async. I could improve the code a bit. 

Also I thought it's always recommended to use async def for activities? Am I wrong on this?

Reply:
>  it would be benefit if we still keep it as async

I don't think so in this case, I think normal `def` would be best

> Also I thought it's always recommended to use async def for activities? Am I wrong on this?

No, actually the opposite, you only use `async def` if you're sure you aren't going to block the thread. We had that async-default stance originally but we've changed documentation and courses and such to encourage just `def` instead.

> I was planning to add the following logic in workflows.py

No need, workflow is good IMO

Reply:
Changed all functions to sync.